src/SphericalCoordinates_TEST.cc

```diff
@@ -383,3 +432,161 @@ TEST(SphericalCoordinatesTest, AssignmentOp)
   math::SphericalCoordinates sc2 = sc1;
   EXPECT_EQ(sc1, sc2);
 }
+
+//////////////////////////////////////////////////
+TEST(SphericalCoordinatesTest, NoHeading)
+{
+  // Default heading
+  auto st = math::SphericalCoordinates::EARTH_WGS84;
+  math::Angle lat(IGN_DTOR(-22.9));
+  math::Angle lon(IGN_DTOR(-43.2));
+  math::Angle heading(0.0);
+  double elev = 0;
+  math::SphericalCoordinates sc(st, lat, lon, elev, heading);
+
+  // Origin matches input
+  auto latLonAlt = sc.SphericalFromLocalPosition({0, 0, 0});
+  EXPECT_DOUBLE_EQ(lat.Degree(), latLonAlt.X());
+  EXPECT_DOUBLE_EQ(lon.Degree(), latLonAlt.Y());
+  EXPECT_DOUBLE_EQ(elev, latLonAlt.Z());
+
+  auto xyzOrigin = sc.LocalFromSphericalPosition(latLonAlt);
+  EXPECT_EQ(math::Vector3d::Zero, xyzOrigin);
+
+  // Check how different lat/lon affect the local position
+
+  // Increase latitude == go North == go +Y
+  {
+    auto xyz = sc.LocalFromSphericalPosition(
+        {lat.Degree() + 1.0, lon.Degree(), elev});
+    EXPECT_NEAR(xyzOrigin.X(), xyz.X(), 1e-6);
+    EXPECT_LT(xyzOrigin.Y(), xyz.Y());
+  }
+
+  // Decrease latitude == go South == go -Y
+  {
+    auto xyz = sc.LocalFromSphericalPosition(
+        {lat.Degree() - 1.0, lon.Degree(), elev});
+    EXPECT_NEAR(xyzOrigin.X(), xyz.X(), 1e-6);
+    EXPECT_GT(xyzOrigin.Y(), xyz.Y());
+  }
+
+  // Increase longitude == go East == go +X (and a bit -Y)
+  {
+    auto xyz = sc.LocalFromSphericalPosition(
+        {lat.Degree(), lon.Degree() + 1.0, elev});
+    EXPECT_LT(xyzOrigin.X(), xyz.X());
+    EXPECT_GT(xyzOrigin.Y(), xyz.Y());
+  }
+
+  // Decrease longitude == go West == go -X (and a bit -Y)
```

Comment:
it seems weird that a change to longitude in either direction causes Y to decrease slightly. I would have thought it would be linearized

Reply:
I was also surprised to see this, I assumed it was an artifact of the projection and didn't look further into it.

Reply:
Ok, so I spent some time looking into this and I think I understand what's happening. This image should help:

![image](https://user-images.githubusercontent.com/5751272/132782393-7e4f6497-2ad2-462a-b913-c8e0833779ee.png)

Note that the ENU plane is tangent to the ellipsoid.

What this test is doing is changing lambda while keeping phi fixed. As you do that, the projection of the point onto the plane will only move on a straight line if you're exactly on the equator. If you're on the northern hemisphere, it goes further North (+Y) and if you're in the southern hemisphere, like on the test, it goes further South (-Y).

Here's a handy way of visualizing it, see how the projection of the blue line is not linear:

![image](https://user-images.githubusercontent.com/5751272/132783031-812b6e65-360e-4678-b64b-d4296c0f6903.png)

https://demonstrations.wolfram.com/TangentPlaneToASphere/